ui/src/app/utils/runtime-utils.tsx

```diff
@@ -278,25 +278,23 @@ const toRuntimeConfig = (runtime: Runtime): RuntimeConfig => {
       machine: findMachineByName(runtime.dataprocConfig.masterMachineType),
       diskSize: runtime.dataprocConfig.masterDiskSize,
       dataprocConfig: runtime.dataprocConfig,
-      pdSize: null
+      pdSize: diskStore.get().persistentDisk != null ? diskStore.get().persistentDisk.size : null
     };
   }
 };
 
-export const getRuntimeConfigDiffs = (oldRuntime: RuntimeConfig, newRuntime: RuntimeConfig): RuntimeDiff[] => {
-  const computeType = newRuntime.computeType;
+export const getRuntimeConfigDiffs = (oldRuntime: RuntimeConfig, newRuntime: RuntimeConfig, gceExists: boolean, pdExists: boolean): RuntimeDiff[] => {
+  // For the compatibility of panel switching between dataproc and running gce without PD
+  const enablePD = serverConfigStore.get().config.enablePersistentDisk;
+  const isCmpDisk = !enablePD || (newRuntime.computeType === ComputeType.Standard) && gceExists && !pdExists || (newRuntime.computeType !== ComputeType.Standard);
```

Comment:
This is really tricky to read - please use some line wrapping and a small improvement I can suggest:

```
const comparePD = enablePD &&
    newRuntime.computeType === ComputeType.Standard &&
    (pdExists || !gceExists);
```